### PR TITLE
fix(test): Preserve default feature flag values when patching in tests

### PR DESCRIPTION
**Pull request by RyanSkonnord**
Change the Feature object, which is used to patch feature flag values in
unit tests, so that it only affects the named flags and preserves
default behavior for the remaining flags. Previously, patching with it
would cause all flags to default to false even if they would default to
true ordinarily.

### CODE REVIEW

**Comment by RyanSkonnord** [comment]
This would solve an issue I'm having in #28621.

**Review by evanpurkhiser — APPROVED** [review]

**Review by wedamija — APPROVED** [review]
looks good to me, although I think this has broken a bunch of tests. We should probably make sure this doesn't cause similar issues in getsentry.

**Comment by RyanSkonnord** [comment]
> looks good to me, although I think this has broken a bunch of tests. We should probably make sure this doesn't cause similar issues in getsentry.

Yep. I'm guessing the broken tests relied on certain flags being implicitly overridden to false. It would be great to make those explicit; hopefully that's not a huge task.

If it turns out to be infeasible, I could maybe add an optional parameter to make flags default globally to false, as before.

👍 to checking on getsentry.

File: tests/sentry/api/endpoints/test_group_integration_details.py [hunk 1]
```diff
@@ -156,7 +156,12 @@ def test_get_feature_disabled(self):
 
         path = f"/api/0/issues/{self.group.id}/integrations/{integration.id}/?action=create"
 
-        with self.feature({"organizations:integrations-issue-basic": False}):
+        with self.feature(
+            {
+                "organizations:integrations-issue-basic": False,
+                "organizations:integrations-issue-sync": False,
+            }
+        ):
```
Comment (RyanSkonnord) [inline tests/sentry/api/endpoints/test_group_integration_details.py:164]: Cases such as this one, where the test calls for more than one flag to be adjusted, make me feel better about investing in the change. Previously, you could have actually gotten away without declaring _either_ as false; `with self.feature({})` would have sufficed.

**Comment by RyanSkonnord** [comment]
I'm approving a fairly large set of changes from the visual snapshot. A few of the integration tests logged a large number of flags that had been defaulting to false, so I believe the reason is that a bunch of flags for unrelated UI changes are now set. Given that the changes under test are purely in the test harness, this should not be risky.

**Pull request merged**